src/main/java/com/aws/greengrass/clientdevices/auth/certificate/infra/BackgroundCertificateRefresh.java

```diff
@@ -122,16 +127,44 @@ public boolean isRunning() {
      * Runs verifyIotCertificate useCase for all the registered client certificate PEMs.
      */
     @Override
-    public void run() {
+    public synchronized void run() {
         if (isNetworkDown()) {
             logger.debug("Network is down - not refreshing certificates");
             return;
         }
 
+        if (!canRun()) {
+            logger.info("Certificates were refreshed recently - retrying later");
```

Comment:
drop to debug, like with network being down?

How often would you think this condition would trigger?

Reply:
Which debug you want me to drop, the one above?.

This condition would trigger based on how choppy the device connection is. If it flickers a lot between up and down this could get triggered multiple times .

Reply:
No, this one which is at info for `!canRun`